### PR TITLE
set autoendpoint to read proper config values

### LINKED ISSUE #186: Fix autoendpoint to use proper config value rather than hostname
autoendpoint defaults to using HOSTNAME over the configuration value. This can lead to problems on EC2 instances where HOSTNAME is already defined by the environment.

Do the right thing and use the endpoint_hostname first, then fail over to hostname.

### PR DESCRIPTION
**Pull request by jrconlin**
* endpoint_hostname and endpoint_scheme now shared values
* default back to previous behaviors

fixes #186
@bbangert r?

### CODE REVIEW

File: autopush/main.py [hunk 1]
```diff
@@ -382,8 +382,9 @@ def endpoint_main(sysargs=None):
     args, parser = _parse_endpoint(sysargs)
     settings = make_settings(
         args,
-        endpoint_scheme="https" if args.ssl_key else "http",
-        endpoint_hostname=args.hostname,
+        endpoint_scheme=args.endpoint_scheme or "https"
+        if args.ssl_key else "http",
```
Comment (bbangert) [inline autopush/main.py]: Nit: the wrapping on this is confusing. Indent?

**Comment by bbangert** [comment]
r+ with nit

File: autopush/main.py (hunk 1 above)
Comment (jrconlin) [inline autopush/main.py]: flake8 was complaining when I had indentation, but I agree that it's confusing. Let me try a few things to see if I can make it better.

Comment (bbangert) [inline autopush/main.py]: Did you want to try a different way, or we could let this in?

**Comment by bbangert** [comment]
LGTM!

**Pull request merged**